docs/advanced/sub-states.md

```diff
@@ -1,9 +1,19 @@
 # Sub Stores
+
 Complex and large state graphs are difficult to manage. Oftentimes we need
 to break these down into sub states that we can manage on a individual
 basis. With NGXS, we can use a concept called sub states to handle this.
 
+## Unidirectional data flow in store
+
+Unidirectional data flow as a pattern is usually mentioned when talking about performance in Angular. The reason why data flows from top to bottom, is because change detection is also always performed from top to bottom for every single component, every single time, starting from the root component. This is awesome, as unidirectional data flow is more predictable than cycles. We always know where the data we use in our views comes from, because it can only result from its component.
```

Comment:
```suggestion
Unidirectional data flow as a pattern is usually mentioned when talking about performance in Angular. The reason why data flows from top to bottom, is because change detection is also always performed from top to bottom for every single component, every single time, starting from the root component. Unidirectional data flow is much easier to debug as it has no side effects unlike the AngularJS's digest cycle. The view is stable throughout a single rendering pass.
```